### PR TITLE
Remove active features stake program v3 and v4

### PR DESCRIPTION
**Pull request by brooksprumo**
Removing features for stake program v3 and v4. The v4 feature was activated back in June 2021 (slot [83376000](https://explorer.solana.com/block/83376000)), so I think they are safe to remove now.

By removing these feature, I was able to clean up some of the staking code paths, which I intend to be working on in subsequent PRs.

### CODE REVIEW

**Review by joncinque — APPROVED** [review]
It's great that the tests still pass even with the old behavior removed! Looks good to me

**Comment by brooksprumo** [comment]
> It's great that the tests still pass even with the old behavior removed!

Well, looks like I missed something... I thought I ran these tests locally first... Investigating.

**Comment by brooksprumo** [comment]
@joncinque Ok, should be fixed now. Sorry about that! I'll re-request a review once the coverage step passes so that I am sure.

File: programs/stake/src/stake_state.rs [hunk 1]
```diff
@@ -3292,24 +3120,24 @@ mod tests {
         let to_account = AccountSharedData::new_ref(1, 0, &system_program::id());
         let to_keyed_account = KeyedAccount::new(&to, false, &to_account);
 
-        // Withdrawing account down to only rent-exempt reserve should succeed before feature, and
-        // fail after
```
Comment (brooksprumo) [inline programs/stake/src/stake_state.rs:3296]: Here's the test that failed. Makes sense after reading the comment 😅

**Review by joncinque — APPROVED** [review]
It seemed too good to be true :laughing:

**Comment by brooksprumo** [comment]
@t-nelson I know [in Discord](https://discord.com/channels/428295358100013066/439194979856809985/933818182131650680) it sounds like _not_ to backport this PR, but does that apply to v1.9 since it's not out on mnb yet?

File: programs/stake/src/stake_state.rs [hunk 2]
```diff
@@ -3292,24 +3120,24 @@ mod tests {
...
+        // Withdrawing account down to minimum balance should succeed
```
Comment (t-nelson) [inline programs/stake/src/stake_state.rs:3123]:
```suggestion
        // Withdrawing account down to minimum delegation should succeed
```

**Review by t-nelson — DISMISSED** [review]
lgtm!  just a comment nit.  Thanks for the cleanup!

**Comment by t-nelson** [comment]
> @t-nelson I know [in Discord](https://discord.com/channels/428295358100013066/439194979856809985/933818182131650680) it sounds like _not_ to backport this PR, but does that apply to v1.9 since it's not out on mnb yet?

Yep. 1.9 is on testnet, which people monitor

File: programs/stake/src/stake_state.rs (hunk 2 above)
Comment (brooksprumo) [reply programs/stake/src/stake_state.rs:3123]: @t-nelson Assuming `MINIMUM_DELEGATION = 1`, this is withdrawing down to `rent_exempt_reserve + MINIMUM_DELEGATION`. That's why I put "balance" here. My understanding is definitely not complete, but wanted to double check; should it be "delegation" instead?

**Pull request merged**